improver/blending/weighted_blend.py

```diff
@@ -472,7 +475,7 @@ def check_compatible_time_points(self, cube: Cube) -> None:
             ValueError : If blending over forecast reference time on a cube
                          with multiple times.
         """
-        if self.timeblending:
+        if self.timeblending or cube.coord("time").ndim > 1:
```

Comment:
As the `time` coordinate is multi-dimensional here, should this function actually be considering the `time_in_local_timezone` coordinate, if it exists? I guess, in principle, there could be multiple values for the `time_in_local_timezone` coordinate?

If this check to skip the time coordinate checking for multi-dimensional time coordinate cubes is preferred, it would be good to update the docstring with this additional clause that multi-dimensional time coordinates won't be checked.

Reply:
Yeah, this is a sensible idea; there should not be multiple values for the `time_in_local_timezone` coordinate when blending the cubes. I've modified the check in the code and the associated test. I've kept the extended doc-string on the test to talk about the multi-dimensional time coordinate.